authd.proto

```diff
@@ -92,7 +92,14 @@ message SAMResponse {
 
 message IARequest {
   string session_id = 1;
-  string authentication_data = 2;
+
+  message AuthenticationData {
+    optional string challenge = 1;
+    optional string code = 2;
```

Comment:
I don’t see the need for code anymore, here. This will be the challenge for now.

Reply:
Ok, I never saw it used, but I saw it in specs, thus the addition.

Reply:
@didrocks wondering, it makes things a bit more verbose to handle, but should we instead use `oneof` for values? Since only one of them should be defined, no?

Reply:
Yeah, I saw it on the spec too this morning and removed it, this is why then I went back to the PR to edit it :)

Very good remark, yeah definitively oneof to validate it wil be beneficial.

Reply:
I've moved to `oneof` but makes things a bit more verbose IMHO, however check if you prefer that instead.